x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/mergers/apply_rule_update.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  RuleResponse,
+  RuleUpdateProps,
+} from '../../../../../../../common/api/detection_engine/model/rule_schema';
+import type { IPrebuiltRuleAssetsClient } from '../../../../prebuilt_rules/logic/rule_assets/prebuilt_rule_assets_client';
+import { applyRuleDefaults } from './apply_rule_defaults';
+import { calculateRuleSource } from './rule_source/calculate_rule_source';
+
+interface ApplyRuleUpdateProps {
+  prebuiltRuleAssetClient: IPrebuiltRuleAssetsClient;
+  existingRule: RuleResponse;
+  ruleUpdate: RuleUpdateProps;
+}
+
+export const applyRuleUpdate = async ({
+  prebuiltRuleAssetClient,
+  existingRule,
+  ruleUpdate,
+}: ApplyRuleUpdateProps): Promise<RuleResponse> => {
+  const nextRule: RuleResponse = {
+    ...applyRuleDefaults(ruleUpdate),
+
+    // Use existing values
+    enabled: ruleUpdate.enabled ?? existingRule.enabled,
+    version: ruleUpdate.version ?? existingRule.version,
+
+    // Always keep existing values for these fields
+    id: existingRule.id,
+    rule_id: existingRule.rule_id,
+    revision: existingRule.revision,
+    immutable: existingRule.immutable,
+    rule_source: existingRule.rule_source,
```

Comment:
We're recalculating this below. Any reason to create it here first?

Reply:
Listing `rule_source` here gives a better sense of what rule fields are ignored during the rule update process. So it is purely for code organization purposes